### PR TITLE
Add prettier and eslint config

### PR DESCRIPTION
**Pull request by alincustura**

### CODE REVIEW

**Review by radubrehar — CHANGES_REQUESTED** [review]
@alincustura please remove yarn.lock

**Comment by alincustura** [comment]
Done!

**Comment by radubrehar** [comment]
@alincustura want to also try setting up the precommit hook to apply prettier formatting and eslint fixes automatically?

**Review by radubrehar — APPROVED** [review]

**Comment by CojocaruDavid** [comment]
> @alincustura want to also try setting up the precommit hook to apply prettier formatting and eslint fixes automatically?

precommit hook will be done in this PR or it will be created another ticket ?

File: frontend/.eslintrc.js
Comment (CojocaruDavid) [inline frontend/.eslintrc.js]: no rules at the moment ?

Comment (alincustura) [reply frontend/.eslintrc.js]: I added two rules. Maybe you suggest some rules. :)

Comment (CojocaruDavid) [reply frontend/.eslintrc.js]: I was thinking at some of them.. Maybe worth checking with the others :) @radubrehar 
```
'react/jsx-wrap-multilines': ['error', {
    arrow: true,
    assignment: true,
    declaration: true,
    return: true
}],
'react/no-unused-state': 'error',
'react/no-will-update-set-state': 'error',
'react/require-render-return': 'error',
'react/self-closing-comp': 'error',
'react/no-multi-comp': ['error', { ignoreStateless: true }],
'react/jsx-no-duplicate-props': ['error', { ignoreCase: true }],
'react/jsx-curly-spacing': ['error', 'never', { allowMultiline: true }],
'react/jsx-equals-spacing': ['error', 'never']
```

More of them can be found **[here](https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules)** with additional documented rules

Comment (radubrehar) [reply frontend/.eslintrc.js]: There are lots are rules @alincustura already added - but they are in the `extends` clause
```
extends: [
    "eslint:recommended",
    "plugin:prettier/recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/eslint-recommended",
  ],
```
All those plugins/configs contain their own rules.
So I would not sweat over adding too many extra rules besides what "recommended" plugins already bring to the game

File: frontend/.eslintrc.js
Comment (radubrehar) [inline frontend/.eslintrc.js]: @alincustura is this rule erroring when there are 4 spaces or is it ensuring there are 4 spaces? I'm not to familiar with how eslint works in this respect. Just make sure it's 2 spaces indenting, to match prettier

File: frontend/.eslintrc.js
Comment (CojocaruDavid) [reply frontend/.eslintrc.js]: Great !

File: frontend/package.json
Comment (radubrehar) [inline frontend/package.json:41]: also add `prettier --write` after eslint, so staged files are also ran through prettier

File: frontend/.eslintrc.js
Comment (alincbiz) [reply frontend/.eslintrc.js]: ok

Comment (alincustura) [reply frontend/.eslintrc.js]: Ok

File: frontend/package.json
Comment (alincustura) [reply frontend/package.json:41]: Ok

**Comment by radubrehar** [comment]
@DavidCojocaru we need you to confirm your review

I still need to give this a round of testing before it gets merged

**Review by CojocaruDavid — APPROVED** [review]
lg 👍

**Review by DavidCojocaru — APPROVED** [review]
👍

**Review by CojocaruDavid — COMMENTED** [review]
@radubrehar Should we request re-review when a commit have been pushed after 2 approvals ? For me makes senes to be so

**Pull request merged**
